packages/typespec-test/test/NetworkAnalytics.Management/generated/typespec-ts/src/api/dataProducts/index.ts

```diff
@@ -493,61 +495,63 @@ export async function _updateDeserialize(
     throw createRestError(result);
   }
 
-  result = result as DataProductsUpdateLogicalResponse;
+  const _result = result as unknown as DataProductsUpdateLogicalResponse;
```

Comment:
here we will do narrowing with two cases:
- [existing] LRO operation because of one LRO operation could have more than one success responses(200, 201, 202) so we narrowed it before this isUnexpected issue happened.
- [newly added] Sub model between default response and non-default response